cirq-core/cirq/sim/simulator_base.py

```diff
@@ -226,13 +286,43 @@ def _run(
 
         measurements: Dict[str, List[np.ndarray]] = {}
         for _ in range(repetitions):
+            copies = {a: a.copy() for a in set(act_on_args.values())}
             all_step_results = self._core_iterator(
                 general_suffix,
-                sim_state=act_on_args.copy(),
+                sim_state={q: copies[a] for q, a in act_on_args.items()},
+                qubits=qubits,
             )
             for step_result in all_step_results:
                 for k, v in step_result.measurements.items():
                     if k not in measurements:
                         measurements[k] = []
                     measurements[k].append(np.array(v, dtype=np.uint8))
         return {k: np.array(v) for k, v in measurements.items()}
+
+    def _create_act_on_args(
+        self,
+        initial_state: Any,
+        qubits: Sequence['cirq.Qid'],
+    ) -> Dict['cirq.Qid', TActOnArgs]:
+        if isinstance(initial_state, dict):
+            return initial_state
+
+        args_map: Dict['cirq.Qid', TActOnArgs] = {}
+        if initial_state is 0 and self._split_untangled_states:
```

Comment:
Any integer `initial_state` is a purely classical state, where the bits of `initial_state` dictate the state of each individual qubit. Based on this, could you update this path to support any integer value of `initial_state`?

Reply:
Nice, will change. We'll need to add that to the docs somewhere so that future simulator implementors are aware of the restriction.